src/src/Helpers/Networking.cpp

```diff
@@ -1682,6 +1682,125 @@ int http_authenticate(const String& logIdentifier,
+    // Visit the URL (https://open-meteo.com/en/docs) and build your personal URL by selecting the location and values you want to receive.
+    // Supported variable kinds are current, hourly, daily!
+    // In rules you can grep the reply by the kind of weather variables with "On Openmeteo#<type> Do ..."
+    // e.g. "On Openmeteo#current Do ..."
+    // Note: hourly and daily results are arrays which can become very long.
+    // Best to make seperate calls. Especially for hourly results.
+
+    if ((httpCode == 200) && equals(host, F("api.open-meteo.com")))
+    {
+      // Length of the response is limited to 5000 characters
+      // ToDo: What would be right size?
+      if (uri.length() > 5000) {
+        addLog(LOG_LEVEL_ERROR, F("Response exceeds 5000 characters"));
+      }
+      else {
+        String str = http.getString();
+
+        // Process URL to get unique keys
+        auto getCombinedParams = [](String url, String paramName) {
+                                   int start = url.indexOf(paramName + "=");
+
+                                   if (start == -1) { return String(""); }
+                                   start += paramName.length() + 1;
+                                   int end = url.indexOf('&', start);
+                                   return (end == -1) ? url.substring(start) : url.substring(start, end);
+                                 };
+
+        // Extract current hourly and daily parameters
+        String currentParams = getCombinedParams(uri, "current");
+        String hourlyParams  = getCombinedParams(uri, "hourly");
+        String dailyParams   = getCombinedParams(uri, "daily");
+
+        auto processAndQueueParams = [](String& params, String& str, String eventName) {
+                                       if (!params.isEmpty()) {
+                                         String keys[20];
+                                         int    keyCount   = 0;
+                                         int    startIndex = 0;
+                                         int    commaIndex = params.indexOf(',');
+
+                                         // Split and add keys to the array
+                                         while (commaIndex != -1) {
+                                           String key = params.substring(startIndex, commaIndex);
+                                           keys[keyCount++] = key;
+                                           startIndex       = commaIndex + 1;
+                                           commaIndex       = params.indexOf(',', startIndex);
+                                         }
+
+                                         // Add the last key
+                                         String lastKey = params.substring(startIndex);
+                                         keys[keyCount++] = lastKey;
+
+                                         String csv              = "";

```

Comment:
No need to initialize with an empty `""` as the default `String` constructor already is an empty string.